Video sync

smu160 opened the pull request:
You can run the code by running `python main.py` in the `examples/video_sync`directory terminal. I utilized a modified version of the PyQt5 in order to write/test/run this code. If you need me to provide my conda environment, I'd be happy to provide you with it. Similarly, if you have any questions/comments/concerns, please let me know.

smu160 commented:
I am printing to stderr using `print(error, file=sys.stderr)`, but I guess this is not available in Python 2. Do you want me to address this? I can simply switch over to `sys.stderr.write()`. Let me know.

oaubert commented:
Thanks for your contribution. Could you do some cleanup on the proposed code:
- provide a more verbose explanation of the code, especially a description of the 3 modules. Just having "video sync" in the pull request is a bit terse to introduce 3 files. You have a client/server architecture, with main.py being the master client, mini_player more basic client and a headless network server. A README.md could be used.
- squash the diff of removal of the vlc.py module (good opportunity to learn about `git rebase -i HEAD~2` :-)
- for you logging issue, please use the logging module instead of print. Basically:

    `import logging`
    `logger = logging.getLogger(__name__)`

at the start of the module, then you can use `logger.info|warn|debug`.

smu160 commented:
I've addressed all the issues you pointed out above, as I understood them. Let me know if anything else needs to be done in regards to that. 

In addition, the figure I created for the README.md made me realize what the exact issue/bug was in regards to videos not always playing/pausing/stopping at the same exact time. I fixed that issue with a `ThreadPoolExecutor` from the `concurrent.futures` [module](https://docs.python.org/3.7/library/concurrent.futures.html#module-concurrent.futures)

oaubert commented:
Great, thanks! One last thing: your files do not have any licensing information, which is important. If you are OK with using GPL licensing, you can copy/paste and adapt the licensing info at the start of any of the existing examples, and put it into each of your .py files.
